.github/workflows/build-test-macos.yml

```diff
@@ -0,0 +1,63 @@
+name: build-and-test-macos
+
+on:
+  push:
+    branches:
+      - master
+
+    paths-ignore:
+      - '**.md'
+      - '.github/**'
+      - '!.github/workflows/build-*'
+      
+
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review]
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref_name }}
+  cancel-in-progress: true
+
+jobs:
+  build-macos:
+    name: Build-macos
+    runs-on: macos-latest
+    strategy:
+      fail-fast: false
+    steps:  
+
+    - name: Set up Go 1.x
+      uses: actions/setup-go@v2
+      with:
+        go-version: ~1.15.15
+      id: go
+
+    - name: Configure git with longpath enabled (for windows)
+      run: git config --global core.longpaths true
+
+    - name: Check out code
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+        submodules: 'true'
+
+    - name: Debug go.mod
+      run: cat go.mod
+
+    - name: Cache build output
+      uses: actions/cache@v2
+      with:
+        path: |
+              ~/Library/Caches/go-build
+              ~/go/pkg/mod
+              /Users/runner/Library/Caches/go-build
+        key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+        restore-keys: |
```

Comment:
We don't need the restore keys since it would cache the old dependencies package.